# Transactional Fraud Detection Analysis

## Week 1: Data Preparation and Initial Exploration

**Student:** Prajwali Teli

**Tools Used:** Python, Pandas, SQLite, Jupyter Notebook

### Objective
To load the transaction dataset, perform initial profiling, identify missing values and duplicates, and analyze the fraud distribution.

In [1]:
# Import Libraries
import pandas as pd
import sqlite3

In [2]:
#Load Dataset
df = pd.read_csv("Synthetic_Financial_Datasets_log.csv")


In [3]:
#Dataset Size
df.shape

(6362620, 11)

In [4]:
#Dataset Information
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 6362620 entries, 0 to 6362619
Data columns (total 11 columns):
 #   Column          Dtype  
---  ------          -----  
 0   step            int64  
 1   type            str    
 2   amount          float64
 3   nameOrig        str    
 4   oldbalanceOrg   float64
 5   newbalanceOrig  float64
 6   nameDest        str    
 7   oldbalanceDest  float64
 8   newbalanceDest  float64
 9   isFraud         int64  
 10  isFlaggedFraud  int64  
dtypes: float64(5), int64(3), str(3)
memory usage: 534.0 MB


In [5]:
#Basic Statistics
df.describe()

,step,amount,oldbalanceOrg,newbalanceOrig,oldbalanceDest,newbalanceDest,isFraud,isFlaggedFraud
count,6.362620e+06,6.362620e+06,6.362620e+06,6.362620e+06,6.362620e+06,6.362620e+06,6.362620e+06,6.362620e+06
mean,2.433972e+02,1.798619e+05,8.338831e+05,8.551137e+05,1.100702e+06,1.224996e+06,1.290820e-03,2.514687e-06
std,1.423320e+02,6.038582e+05,2.888243e+06,2.924049e+06,3.399180e+06,3.674129e+06,3.590480e-02,1.585775e-03
min,1.000000e+00,0.000000e+00,0.000000e+00,0.000000e+00,0.000000e+00,0.000000e+00,0.000000e+00,0.000000e+00
25%,1.560000e+02,1.338957e+04,0.000000e+00,0.000000e+00,0.000000e+00,0.000000e+00,0.000000e+00,0.000000e+00
50%,2.390000e+02,7.487194e+04,1.420800e+04,0.000000e+00,1.327057e+05,2.146614e+05,0.000000e+00,0.000000e+00
75%,3.350000e+02,2.087215e+05,1.073152e+05,1.442584e+05,9.430367e+05,1.111909e+06,0.000000e+00,0.000000e+00
max,7.430000e+02,9.244552e+07,5.958504e+07,4.958504e+07,3.560159e+08,3.561793e+08,1.000000e+00,1.000000e+00


In [6]:
#Missing Values
df.isnull().sum()

step              0
type              0
amount            0
nameOrig          0
oldbalanceOrg     0
newbalanceOrig    0
nameDest          0
oldbalanceDest    0
newbalanceDest    0
isFraud           0
isFlaggedFraud    0
dtype: int64

In [7]:
#Duplicate Records
df.duplicated().sum()

np.int64(0)

In [8]:
#Fraud Distribution
df["isFraud"].value_counts()

isFraud
0    6354407
1       8213
Name: count, dtype: int64

In [9]:
#Fraud Rate
fraud_percentage = df["isFraud"].mean() * 100
print(fraud_percentage)

0.12908204481801522


In [10]:
#Fraud by Transaction Type
df.groupby("type")["isFraud"].sum()

type
CASH_IN        0
CASH_OUT    4116
DEBIT          0
PAYMENT        0
TRANSFER    4097
Name: isFraud, dtype: int64

In [11]:
#Fraud Amount Analysis
fraud_df = df[df["isFraud"] == 1]

fraud_df["amount"].describe()

count    8.213000e+03
mean     1.467967e+06
std      2.404253e+06
min      0.000000e+00
25%      1.270913e+05
50%      4.414234e+05
75%      1.517771e+06
max      1.000000e+07
Name: amount, dtype: float64

In [12]:
#Transaction Type Consistency
df["type"].value_counts()

type
CASH_OUT    2237500
PAYMENT     2151495
CASH_IN     1399284
TRANSFER     532909
DEBIT         41432
Name: count, dtype: int64

In [ ]:
##SQL Analysis
#Results
1.Total transactions → 6,362,620
2.Fraud transactions → 8,213
3.CASH_OUT fraud → 4,116
4.TRANSFER fraud → 4,097

##Week 1 Summary

The transaction dataset contains 6,362,620 records and 11 columns. No missing values or duplicate records were identified. Fraudulent transactions account for approximately 0.13% of all transactions, indicating a highly imbalanced dataset. Fraud cases were concentrated in CASH_OUT and TRANSFER transaction types. SQL was also used to verify transaction counts and fraud patterns.